rust/datafusion/src/physical_plan/hash_join.rs

```diff
@@ -143,13 +143,10 @@ impl ExecutionPlan for HashJoinExec {
         self.right.output_partitioning()
     }
 
-    async fn execute(&self, partition: usize) -> Result<SendableRecordBatchStream> {
+    async fn execute(&self) -> Result<Vec<SendableRecordBatchStream>> {
```

Comment:
Ah, I had missed this detail on the earlier versions of this PR -- that each invocation of execute for the probe side needed to call execute on the build side.

An alternate implementation would be to store the hash table on `self` (in a Arc<Mutex<..>>) and reuse it on subsequent invocations.

I like the pattern of this PR where each `execute` method is invoked once and then the async streams are used to coordinate the threading / execution. 

👍